File: 20A40/cir-agus/EJER_36/Administrador.py
from Empleado import Empleado

class Administrador(Empleado):
    
    def __init__(self,nombre,salario,fecha,deptoC, bonoC) -> None:
        self.__depto = deptoC
        self.__bono = bonoC
        super().__init__(nombre,salario,fecha)   

    def setDepto(self,dept) -> bool:
        if len(dept) > 0:
            self.__depto = dept
            return True
        else: return False
    
    def setBono(self,bono) -> bool:
        if bono > 0:
            self.__bono = bono
            return True
        else: return False

    def getDepto(self) -> str: return self.__depto
    def getBono(self)  -> float: return self.__bono

    def toString(self) -> str:
        resul = super().toString()
        
        resultado = resul + f"\nDepartamento {self.__depto}\n"
        resultado += f"Bono: {self.__bono}"
        return resultado

File: 41A60/cir-agus/EJER_43-Java/Administrador.java
public class Administrador extends Empleado{
    private String Departamento;
    private double Bono;

    public Administrador(){}
    
    public Administrador(
        String name, double sal, String bd, String dept, double bono
           )
    {
        super(name,sal,bd);
        this.Departamento = dept;
        this.Bono = bono;
    }

    public boolean setDepartamento(String dept){
        if(dept != null && !dept.equals("")){
            this.Departamento = dept;
            return true;
        } else return false;
    }

    public boolean setBono(double bono){
        if(bono > 0){
            this.Bono = bono;
            return true;
        }else return false;
    }

    public String getDepartamento(){return Departamento;}
    public double getBono(){ return Bono; }

    public String toString(){ //Redifinacion del metodo toString()
        String toM = super.toString();
    
        String cadString = toM +"Departamento: "+Departamento+"\n";
        cadString += "Bono "+Bono+"\n";
        return cadString;
      }

    
}
